src/wallet.js

```diff
@@ -286,7 +286,7 @@ const wallet = {
       this.setWalletData(dataJson);
       storage.setStore(oldStore);
 
-      this.getTxHistory(addresses).then((history) => {
+      this.getTxHistory(addresses, connection, store).then((history) => {
```

Comment:
As per my comment on `getTxHistory`, we seem to be updating the storage there already. I think we either do everything there or everything here.

Reply:
It's doing everything there right now.